doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,435 @@
+
+Decrypts an encrypted string using the algorithm and a key string.
+
+.. rubric:: Returns
+
+A plaintext as a string.
+
+.. rubric:: Parameters
+
+The following are the function's parameters:
+
+* algorithm - encryption algorithm used to encrypt the string. Supports `RSA`.
+
+* key_str - a string in the PEM format. The key string must be the following:
+
+  - Valid
+
+  - Public or private key string that corresponds with the private or public key string used with the :ref:`asymmetric_encrypt<asym-encrypt>` function.
+
+.. _asym-derive:
+
+asymmetric_derive(*pub_key_str, priv_key_str*)
+------------------------------------------------------------
+
+Derives a symmetric key using a public key generated on one side and a private key generated on another.
+
+.. rubric:: Returns
+
+A key as a binary string.
+
+.. rubric:: Parameters
+
+The ``pub_key_str`` must be a public key in the PEM format and generated using the Diffie-Hellman (DH) algorithm.
+
+The ``priv_key_str`` must be a private key in the PEM format and generated using the Diffie-Hellman (DH) algorithm.
+
+.. _asym-encrypt:
+
+asymmetric_encrypt(*algorithm, str, key_str*)
+--------------------------------------------------------------
+
+Encrypts a string using the algorithm and a key string.
+
+.. rubric:: Returns
+
+A ciphertext as a binary string.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - encryption algorithm used to encrypt the string. Supports `RSA`.
+
+* str - measured in bytes, the length of the string must not be greater than the key_str modulus length in bytes - 11
+
+* key_str - a key (either private or public) in the PEM format
+
+.. _asym-sign:
+
+asymmetric_sign(*algorithm, digest_str, priv_key_str, digest_type*)
+------------------------------------------------------------------------
+
+Signs a digest string using a private key string.
+
+.. rubric:: Returns
+
+A signature is a binary string.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA' or 'DSA'.
+
+* digest_str - the digest binary string that is signed. The string can be generated by invoking :ref:`create-digest<create-digest>`.
+
+* priv_key_str - the private key used to sign the digest string. The key must be in the PEM format.
+
+* digest_type - the supported values are listed in the :ref:`create-digest<create-digest>` digest type table.
+
+
+.. _asym-verify:
+
+asymmetric_verify(*algorithm, digest_str, sig_str, pub_key_str, digest_type*)
+--------------------------------------------------------------------------------
+
+Verifies whether the signature string matches the digest string.
+
+.. rubric:: Returns
+
+A ``1`` (success) or a ``0`` (failure).
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - the supported values are 'RSA' or 'DSA'.
+
+* digest_str - the digest binary string and this string can be generated by invoking :ref:`create-digest<create-digest>`.
```

Comment:
Please replace "-" with an underscore "_" in the function name **create_digest**

Reply:
all dashes changed to underscores